modules/KalturaSupport/components/streamSelectorUtils.js

```diff
@@ -74,30 +74,41 @@
         },
         getStreams: function () {
             var _this = this;
-            var requestObject = [];
-            requestObject.push({
-                'service': 'baseEntry',
-                'action': 'list',
-                'filter:objectType': 'KalturaBaseEntryFilter',
-                'filter:typeEqual': 1,
-                'filter:parentEntryIdEqual': this.getPlayer().kentryid
-            });
-
-            var i = 0;
-            var maxNumOfStream = this.getConfig("maxNumOfStream");
-            for (i; i < maxNumOfStream; i++) {
-                requestObject.push({
-                    'service': 'flavorAsset',
+            var requestObject = [
+                {
+                    'service': 'baseEntry',
                     'action': 'list',
-                    'filter:entryIdEqual': '{1:result:objects:' + i + ':id}'
-                });
-            }
-
-            // do the api request
-            this.getKalturaClient().doRequest(requestObject, function (data) {
+                    'filter:objectType': 'KalturaBaseEntryFilter',
+                    'filter:typeEqual': 1,
+                    'filter:parentEntryIdEqual': this.getPlayer().kentryid
+                }
+            ];
+            var responceArray = [];
+            // get flavorAssets id
+            _this.getKalturaClient().doRequest(requestObject, function (data) {
                 // Validate result
-                if (data && _this.isValidResult(data[0] && data[0].totalCount > 0)) {
-                    _this.createStreamList(data);
+                if (data && _this.isValidResult(data && data.totalCount > 0)) {
+                    responceArray.push(data)
+                    var i = 0;
+                    var maxNumOfStream = _this.getConfig("maxNumOfStream");
+                    var flavorAssetRequestObject = [];
+                    for (i; i < data.totalCount && i < maxNumOfStream; i++) {
```

Comment:
@semarche-kaltura 
consider use 
maxNumOfStream = math.Max(maxNumOfStream,data.totalCount) and then in the for-loop use a simpler expression

Reply:
done